apps/kg/train_pytorch.py

```diff
@@ -57,6 +118,9 @@ def train(args, model, train_sampler, valid_samplers=None, rank=0, rel_parts=Non
         sample_time += time.time() - start1
         args.step = step
 
+        if client is not None:
+            model.pull_model(client, pos_g, neg_g)
```

Comment:
is this asynchronous?

Reply:
Pull is not async but I optimized it by overlapping the local computation and communication.